make scheduler err handler func clearly and test it

fxmumu opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

make the code clearly and read easily in scheduler err handler function.
test for scheduler error node not found.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:



**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

Commit:
Reconcile NoExecute Taint

Commit:
Use e2epod.WaitForPodTerminatedInNamespace directly

WaitForPod*() are just wrapper functions for e2epod package, and they
made an invalid dependency to sub e2e framework from the core framework.
So this replaces WaitForPodTerminated() with the e2epod function.

aojea committed:
cluster: ipvs conntrack module vs kernel version

We should use 'nf_conntrack' instead of 'nf_conntrack_ipv4'
for linux kernel >= 4.19

tanjunchen committed:
WaitForPodReady is simply wrapper functions for e2epod package,
and they made an invalid dependency to sub e2e framework from the core framework.

So we can use e2epod.WaitTimeoutForPodReadyInNamespace to remove invalid dependency.

The main purpose of this pr is to handle the framework core package dependency subpackage pod.

aojea committed:
bump k8s.io utils

Fix a bug in k8s.io/utils/net.GetIndexedIP() that wasn't working
with IPv6 addresses with leading zeros.

Commit:
Simplify if else branch in schduler error handler function, and add test for node not found error.

fxmumu commented:
/sig scheduling

fxmumu commented:
/ptal

fxmumu commented:
/auto-cc

draveness commented on pkg/scheduler/factory.go:
```diff
@@ -455,28 +455,28 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
 func MakeDefaultErrorFunc(client clientset.Interface, podQueue internalqueue.SchedulingQueue, schedulerCache internalcache.Cache) func(*framework.PodInfo, error) {
 	return func(podInfo *framework.PodInfo, err error) {
 		pod := podInfo.Pod
+		_, isFitError := err.(*core.FitError)
```
This is not a good one IMO.

fxmumu replied:
Thx, I will fix it later.

fxmumu replied:
Like this?
```
else if _, ok := err.(*core.FitError); ok {
}
```
I had asked a friend, he suggested me.

draveness replied:
> Like this?
> 
> ```
> else if _, ok := err.(*core.FitError); ok {
> }
> ```
> 
> I had asked a friend, he suggested me.

Yes

fxmumu replied:
A shorter lifecycle var is welcome, I got it. Thx

fxmumu replied:
> > Like this?
> > ```
> > else if _, ok := err.(*core.FitError); ok {
> > }
> > ```
> > 
> > 
> > I had asked a friend, he suggested me.
> 
> Yes

A shorter lifecycle var is welcome, I got it. Thx

draveness commented on pkg/scheduler/factory.go:
```diff
@@ -455,28 +455,28 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
-						if err := schedulerCache.RemoveNode(&node); err != nil {
-							klog.V(4).Infof("Node %q is not found; failed to remove it from the cache.", node.Name)
-						}
+		} else if isFitError {
```
but we could have this structure to remove the nested if statement

fxmumu replied:
> but we could have this structure to remove the nested if statement

what can I do for this comment? or just resolve it directly?

draveness reviewed (commented):
/ok-to-test

draveness commented on pkg/scheduler/factory.go:
```diff
@@ -455,28 +455,27 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
-						}
+		} else if _, ok := err.(*core.FitError); ok {
+			klog.V(2).Infof("Unable to schedule %v/%v: no fit: %v; waiting", pod.Namespace, pod.Name, err)
+		} else if resourceNotFound {
```
```suggestion
		} else if apierrors.IsNotFound(err) {
```

fxmumu replied:
OK, I got it. Actually I want to express `resource` not found. Of course, the func `IsNotFound` had expressed that. So I will accept you suggestion and Thx.
Something wrong with bazel,  it seems that scheduler pkg test dependencies not sync to bazel BUILD file, I'm trying to resolve it.

fxmumu commented:
> @fxmumu: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-node-e2e-containerd	[998b8d1](https://github.com/kubernetes/kubernetes/commit/998b8d11addfed15b10bd5527d7b3f61200e3d09)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/89375/pull-kubernetes-node-e2e-containerd/1242638667164422144/)	`/test pull-kubernetes-node-e2e-containerd`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=89375). [Your PR dashboard](https://gubernator.k8s.io/pr/fxmumu). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

fxmumu committed:
Remove nested if statement and test it

remove nested if statement in scheduler err handler.
test for the scheduler err that node not found.

fxmumu commented:
re PR later
